File: Benchmarking/stats/euler22/euler22.java
import java.util.*;
import java.io.*;

public class euler22
{
	public static void main(String[] args)
	{
		String file = args[0];
		try
		(
			InputStream fis = new FileInputStream(file);
			InputStreamReader isr = new InputStreamReader(fis);
			BufferedReader br = new BufferedReader(isr);
		)
		{
			String line = br.readLine();
			br.close();
			isr.close();
			fis.close();
			calc(letterValues(), namesSorted(line));
		}
		catch(Exception e)
		{
			System.out.println(e);
		}
	}

	private static ArrayList<String> namesSorted(String line)
	{
		String line_new = line.replace((char) 22, (char) 0);
		ArrayList<String> array = new ArrayList<String>(Arrays.asList(line_new.split(",")));
		Collections.sort(array);
		return array;
	}

	private static HashMap<String, Integer> letterValues()
	{
		String[] letters = new String[]{"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"};
		HashMap<String, Integer> mapo = new HashMap<>();
		int valueLetter = 1;
		for(String s : letters)
		{
			mapo.put(s, valueLetter);
			//valueLetter += 1;
		}
		return mapo;
	}

	private static int calc(HashMap<String, Integer> mapo, ArrayList<String> names_sorted)
	{
		int value = 0;
		int mapo_index = 1;
		for(String word : names_sorted)
		{
			int value_word = 0;
			for(String ch : word.split(""))
			{
				value_word += mapo.getOrDefault(ch, 0);
			}
			value += value_word * mapo_index;
			mapo_index += 1;
		}
		return value;
	}
}

File: Benchmarking/stats/euler22/euler22.py
import sys

path = sys.argv[1]
file = open(path)

def namesSorted(file):
  line = file.readline()
  line_new = str.replace(line, '"', "")
  array = line_new.split(',')
  file.close()
  names = sorted(array)
  return names

def letterValues():
  letters = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]
  mapo = {}
  valueLetter = 1
  for a in letters:
    mapo[a] = valueLetter
    #valueLetter += 1 #For correct values
  return mapo

def calc(mapo, names_sorted):
  value = 0
  mapo_index = 1
  for word in names_sorted:
    value_word = 0
    for ch in word:
      value_word += mapo.get(str(ch), 0)
    value += value_word * mapo_index
    mapo_index += 1
  return value

calc(letterValues(), namesSorted(file))
